Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Extensions/ApiGatewayResponseExtensions.cs

```diff
@@ -0,0 +1,317 @@
+﻿namespace Amazon.Lambda.TestTool.Extensions;
+
+using Amazon.Lambda.APIGatewayEvents;
+using Amazon.Lambda.TestTool.Models;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Primitives;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection.PortableExecutable;
+using System.Text;
+using System.Text.Json;
+
+/// <summary>
+/// Provides extension methods for converting API Gateway responses to HttpResponse objects.
+/// </summary>
+public static class ApiGatewayResponseExtensions
+{
+
+    private const string InternalServerErrorMessage = "{\"message\":\"Internal Server Error\"}";
+
+    /// <summary>
+    /// Converts an APIGatewayProxyResponse to an HttpResponse.
+    /// </summary>
+    /// <param name="apiResponse">The API Gateway proxy response to convert.</param>
+    /// <returns>An HttpResponse representing the API Gateway response.</returns>
+    public static HttpResponse ToHttpResponse(this APIGatewayProxyResponse apiResponse, ApiGatewayEmulatorMode emulatorMode)
```

Comment:
Returning an `HttpResponse` will make it hard for us to call this extension method from the minimal API and then return the new `HttpResponse`. The `HttpResponse` is tightly coupled with the `HttpContext` and cannot be updated as it does not have a setter. We will need to update the existing `HttpResponse` in-place.
The method should look like:
```
public static void ApplyToHttpResponse(
    this APIGatewayProxyResponse apiResponse, 
    HttpContext context, 
    ApiGatewayEmulatorMode emulatorMode)
```

Reply:
fixed